packages/calcite-components/src/components/accordion/accordion.tsx

```diff
@@ -128,18 +128,7 @@ export class Accordion {
   //
   //--------------------------------------------------------------------------
 
-  mutationObserver = createObserver("mutation", (mutationList) => {
-    let addedAccordionItem: HTMLCalciteAccordionItemElement;
-    for (const mutation of mutationList) {
-      if (mutation.type === "childList") {
-        const addedNodes = Array.from(mutation.addedNodes);
-        addedAccordionItem = addedNodes.find(
-          (node) => node instanceof HTMLCalciteAccordionItemElement
-        ) as HTMLCalciteAccordionItemElement;
-      }
-    }
-    this.updateAccordionItems(addedAccordionItem);
```

Comment:
Hmm idk how I feel about this. Removing this part will have a performance impact because now the DOM is being queried on every mutation, whereas before it was only being queried when nodes were added. 

`querySelectorAll` is a pretty intensive task, especially for something like accordion which can have a yuuge amount of child nodes.

Reply:
Do we really need to query the DOM at all? We should be able to get the added/removed/target nodes from the mutation observer right

Reply:
updateAccordionItems() runs also on watcher for a number of props that are set by parents on children. We need to keep track of items for updateAccordionItems to map set props/prop changes onto them. Looks like at this point that's all updateAccordionItems is doing, as we no longer keep track of position, or items being added because of the query.

Mutation observer track additions, removals, and attribute changes, right? If that's the case why do we even need a watcher? We could just run the update function to sync props on component load in addition to the observer, but we'd need to query still.

Reply:
Yeah now that the constraint of only querying when the mutation included addedNodes is removed, I'm pretty sure the watchers are redundant. 

However I'm still concerned about using querySelectorAll on every mutation. There has to be a better way to do this that doesn't have the potential to explode someone's computer. Found this explaination after some googling: https://stackoverflow.com/a/39332340

Reply:
Nevermind, it looks like the MutationObserver isn't set up to observe attributes so I think we would still need to watchers

Reply:
Also it's not watching the whole subtree so the performance hit won't be too bad, I guess. Does the current logic cover nested accordions or accordion items?